app/routes/__index/certificate/index.tsx

```diff
@@ -1,57 +1,80 @@
 import { Flex, Center } from '@chakra-ui/react';
-import type { Certificate } from '@prisma/client';
-import { useState } from 'react';
+import type { LoaderArgs, ActionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import { useTypedLoaderData } from 'remix-typedjson';
+import { useRevalidator } from '@remix-run/react';
+import { useInterval } from 'react-use';
 
-import undrawSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import { requireUser } from '~/session.server';
+import pendingSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import failedSvg from '~/assets/undraw_cancel_re_pkdm.svg';
 import Loading from '~/components/display-page';
 import CertificateAvailable from '~/components/certificate/certificate-available';
 import CertificateRequestView from '~/components/certificate/certificate-request';
 import { useUser } from '~/utils';
+import { getCertificateStatusByUsername } from '~/models/certificate.server';
+import { addCertRequest } from '~/queues/certificate/certificate-flow.server';
 
-export default function CertificateIndexRoute() {
-  const user = useUser();
-  const [certificateRequested, setCertificateRequested] = useState(false);
-  const [loading, setLoading] = useState(false);
+export const loader = async ({ request }: LoaderArgs) => {
+  const user = await requireUser(request);
 
-  const currentDate: Date = new Date(2023, 2);
+  return await getCertificateStatusByUsername(user.username);
```

Comment:
Get rid of `await` here.  Also, since a certificate will include dates, should we use @Myrfion's trick with returning `typedjson`?  See for example https://github.com/DevelopingSpace/starchart/blob/d4b1db4e5eb873d1da4e34f29d71bdaabb8b2400/app/routes/__index/dns-records/%24dnsRecordId.tsx#L32